.github/workflows/test-build-deploy.yml

```diff
@@ -184,9 +196,9 @@ jobs:
       - name: Upgrade golang
         uses: actions/setup-go@v2
         with:
-          go-version: 1.19.3
+          go-version: 1.20.1
       - name: Checkout repository
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
```

Comment:
Do we need to do `run: git config --global --add safe.directory /__w/mimir/mimir` magic here as well?

Reply:
@pstibrany though evidently the build works, so I'm unsure whether or why we need `git config --global --add safe.directory` at all.

Reply:
When I look at diff in this PR, I see that we run `git config --global --add safe.directory /__w/mimir/mimir` after some checkouts, but not everywhere. When do we need to do that and when can we skip that step?

Reply:
> @pstibrany though evidently the build works, so I'm unsure whether or why we need `git config --global --add safe.directory` at all.

If it's not required, let's remove it :)

Reply:
@pstibrany I can try to experiment in a draft PR of mine, to see which changes are necessary.

Reply:
From experimenting, I find that the checkout@v3 [`set-safe-directory` option](https://github.com/actions/checkout#usage) doesn't work. Maybe it configures the Git config of a different user than the build runs as? That's why Ying's workaround is necessary I would say.

Reply:
> Do we need to do run: git config --global --add safe.directory /__w/mimir/mimir magic here as well?

Since it doesn't fail (for some reason), I would say it's not necessary?

Reply:
> Do we need to do `run: git config --global --add safe.directory /__w/mimir/mimir` magic here as well?

To answer my question, we only need it when we want to run `git` command from our steps, and action runs inside the container. So not needed here.